src/gui/editorwidgets/qgsdefaultsearchwidgetwrapper.cpp

```diff
@@ -55,19 +56,29 @@ void QgsDefaultSearchWidgetWrapper::setCaseString( int caseSensitiveCheckState )
 
 void QgsDefaultSearchWidgetWrapper::setExpression( const QString &expression )
 {
-  const QMetaType::Type fldType = layer()->fields().at( mFieldIdx ).type();
-  const bool numeric = ( fldType == QMetaType::Type::Int || fldType == QMetaType::Type::Double || fldType == QMetaType::Type::LongLong );
-
-  QString exp = expression;
   const QString nullValue = QgsApplication::nullRepresentation();
   const QString fieldName = layer()->fields().at( mFieldIdx ).name();
   QString str;
-  if ( exp == nullValue )
+  if ( expression == nullValue )
   {
     str = QStringLiteral( "%1 IS NULL" ).arg( QgsExpression::quotedColumnRef( fieldName ) );
   }
   else
   {
+    QString exp = expression;
+    const QMetaType::Type fldType = layer()->fields().at( mFieldIdx ).type();
```

Comment:
Have you considered using `QgsVariantUtils::isNumericType(QMetaType::Type metaType )` ?

Reply:
No. Because it was solved this way before I didn't wanted to change to much on existing code. But since I added other types I think isNumericType would be appropriate. Thanks.